iFixit/react-commerce
# Upgrade strapi version

I had implemented this before #555 was opened. This one also converts the project to use Typescript.

cc @masonmcelvain 

## Changelog

- Upgrade Strapi to v4.3.0
- Refactor `backend` folder to use Typescript (excluding the seeding plugin that will be addressed in a follow-up PR)

## QA

1. Open Vercel preview
2. Perform a smoke test to verify that product list page still renders correctly
3. Open [Govinor preview](https://upgrade-strapi-version.govinor.com/admin) of this branch
4. Navigate to settings and verify that the version is `4.3.0`
5. Change some data and verify that changes are reflected in the frontend (note that CDN caching might affect this step, so you might need a few refreshes to get fresh data on the frontend)

<hr/>

⚠️ ⚠️ ⚠️ After merging this, we need to re-deploy the production instance of Strapi as well

cc @sterlinghirsh
